python/src/transforms.cpp

```diff
@@ -888,8 +1332,10 @@ void init_transforms(nb::module_& m) {
          const nb::object& outputs,
          bool shapeless) {
         //  Try to get the name
-        auto n = fun.attr("__name__");
-        auto name = n.is_none() ? "compiled" : nb::cast<std::string>(n);
+        auto n =
+            nb::hasattr(fun, "__name__") ? fun.attr("__name__") : nb::none();
+        auto name = n.is_none() ? "compiled"
+                                : nb::cast<std::string>(fun.attr("__name__"));
```

Comment:
Nice fix. Just leaving this as a todo for later. Probably should add this to the the custom function / checkpointed function at some point (?). Maybe we can refactor it a bit.